src/time.rs

```diff
@@ -255,4 +340,68 @@ impl Time {
         total_seconds += self.second as u32;
         total_seconds
     }
+
+    /// Clone the time and set a new timezone offset.
+    ///
+    /// The returned time will represent a different point in time since the timezone offset is changed without
+    /// modifying the time. See [Time::in_timezone] for alternative behaviour.
+    ///
+    /// # Arguments
+    ///
+    /// * `tz_offset` - optional timezone offset in seconds.
+    ///
+    /// This method will return `Err(ParseError::OutOfRangeTz)` if `abs(tz_offset)` is not less than 24 hours `86_400`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use speedate::Time;
+    ///
+    /// let t1 = Time::parse_str("12:13:14Z").unwrap();
+    ///
+    /// let t2 = t1.with_timezone_offset(Some(-8 * 3600)).unwrap();
+    /// assert_eq!(t2.to_string(), "12:13:14-08:00");
+    /// ```
+    pub fn with_timezone_offset(&self, tz_offset: Option<i32>) -> Result<Self, ParseError> {
+        if let Some(offset_val) = tz_offset {
+            if offset_val.abs() >= 24 * 3600 {
+                return Err(ParseError::OutOfRangeTz);
+            }
+        }
+        let mut time = self.clone();
+        time.tz_offset = tz_offset;
+        Ok(time)
+    }
+
+    /// Create a new time in a different timezone.
+    /// See [Time::with_timezone_offset] for alternative behaviour.
+    ///
+    /// The time must have an offset, otherwise a `ParseError::TzRequired` error is returned.
+    ///
+    /// # Arguments
+    ///
+    /// * `tz_offset` - new timezone offset in seconds.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use speedate::Time;
+    ///
+    /// let t1 = Time::parse_str("15:00:00Z").unwrap();
+    ///
+    /// let t2 = t1.in_timezone(7200).unwrap();
+    // / assert_eq!(t2.to_string(), "17:00:00+02:00");
+    /// ```
+    pub fn in_timezone(&self, tz_offset: i32) -> Result<Self, ParseError> {
+        if tz_offset.abs() >= 24 * 3600 || tz_offset.abs() > self.total_seconds() as i32 {
+            Err(ParseError::OutOfRangeTz)
+        } else if let Some(current_offset) = self.tz_offset {
+            let seconds = self.total_seconds() as i32 + (tz_offset - current_offset);
+            let mut t = Self::from_timestamp(seconds.try_into().unwrap(), self.microsecond)?;
```

Comment:
This was somewhat unclear to me, since if the total_seconds is less than the offset seconds, the conversion here will fail.
That's why I added this check: `if tz_offset.abs() >= 24 * 3600 || tz_offset.abs() > self.total_seconds() as i32 ` but still looks a big off.